### PR TITLE
Add the Missing Gates for Conversion

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
Adds the following missing gates for the conversion -

1. `CH` -> `CHGate`
2. `CPhase` -> `CPhaseGate`
3. `CY` -> `CYGate`
4. `CCZ` -> `CCZGate`
5. `ECR` -> `ECRGate`
6. `Barrier` -> `Barrier`
7. `GlobalPhase` -> `GlobalPhaseGate`*

*(Qiskit follows the inverse convention)

### CODE REVIEW

**Comment by obliviateandsurrender** [comment]
[sc-56127]

File: pennylane_qiskit/qiskit_device.py [hunk 1]
```diff
@@ -74,6 +74,13 @@
     "Adjoint(S)": lib.SdgGate,
     "Adjoint(T)": lib.TdgGate,
     "Adjoint(SX)": lib.SXdgGate,
+    "CY": lib.CYGate,
+    "CH": lib.CHGate,
+    "CPhase": lib.CPhaseGate,
+    "CCZ": lib.CCZGate,
+    "ECR": lib.ECRGate,
+    "Barrier": lib.Barrier,
+    "Adjoint(GlobalPhase)": lib.GlobalPhaseGate,
```
Comment (albi3ro) [inline pennylane_qiskit/qiskit_device.py:83]: Is there tests for these?

Comment (obliviateandsurrender) [reply pennylane_qiskit/qiskit_device.py:83]: My understanding was that these gets checked with the `integration-tests`, which runs it with  the `devices/tests/test_gates.py` from the PennyLane.

Comment (albi3ro) [reply pennylane_qiskit/qiskit_device.py:83]: Might be best to follow up with an additional plugin-specific test too anyway.

**Review by lillian542 — APPROVED** [review]

File: pennylane_qiskit/qiskit_device.py (hunk 1 above)
Comment (obliviateandsurrender) [reply pennylane_qiskit/qiskit_device.py:83]: Added it!

**Pull request merged**